atkinson/pungi/depends.py

```diff
@@ -0,0 +1,308 @@
+
+        self.filename = filename
+
+        if not self.config or self.config == {} or \
+                'build_sources' not in self.config:
+            return
+
+        srcs = self.config['build_sources']
+        if version not in srcs:
+            raise ValueError('Invalid version: {0}, expected one of {1}'.format(
+                version,
+                [ver for ver in srcs]))
+
+        repos = []
+        for idx in range(len(srcs[version])):
+            repos.append(Repo(str(idx), srcs[version][idx]))
+
+        # TODO: rpmreq supporting passdown of logger
+        # e.g. if self.provided_log:
+        #          (add logger=self.log to kwargs)
+        met_deps, wrong_version, unmet_deps = \
+            build_requires(filename, repos)
+
+        self.met = transmogrify_met(met_deps)
+        self.wrong_version = transmogrify_unmet(wrong_version)
+        self.unmet = transmogrify_unmet(unmet_deps)
+
+        if 'logger' in kwargs:
+            kwargs['logger'].info(str(self))
+
+    def __str__(self):
+        """Report useful information about ourself."""
+        fname = os.path.basename(self.filename)
+        str_format = '{0}: {1} met, {2} incorrect, {3} missing'
+        return str_format.format(fname, len(self.met), len(self.wrong_version), len(self.unmet))
+
+
+def transmogrify_met(met_deps):
+    """
+    Convert a Hawkey dependency into a simple dict
+
+    This is generally only used by DependencySet, but could be used
+    elsewhere
+    """
+    if not isinstance(met_deps, list):
+        raise ValueError('met_deps is not a list')
+    ret = []
+    for dep in met_deps:
+        info = {'component': componentize(dep.sourcerpm),
+                'epoch': dep.epoch,
+                'name': dep.name,
+                'comparison': '==',   # This is the version provided
+                'version': dep.version,
+                'release': dep.release}
+        ret.append(info)
+    return ret
+
+
+def transmogrify_unmet(unmet_deps):
+    """
+    Convert list of string or repo.Depends to our simple dict format
+
+    This is generally only used by DependencySet, but could be used
+    elsewhere
+    """
+    if not isinstance(unmet_deps, list):
+        raise ValueError('unmet_deps is not a list')
+    ret = []
+    for dep in unmet_deps:
+        info = rpmdep_to_dep(str(dep))
+        ret.append(info)
+    return ret
+
+
+def dep_to_tuple(dep):
+    """Convert one of our dependencies into an RPM-style tuple"""
+    if not isinstance(dep, dict):
+        raise ValueError('dep is not a dict')
+    if 'version' not in dep:
+        raise ValueError('Cannot convert {0}: version missing'.format(str(dep)))
+    epoch = 0
+    version = None
+    release = None
+    if 'epoch' in dep:
+        epoch = int(dep['epoch'])
+    version = dep['version']
+    if 'release' in dep:
+        release = dep['release']
+    return (epoch, version, release)
+
+
+def dep_to_nevr(dep):
```

Comment:
These could probably remain outside the object, making them more reusable (and possibly moving them to toolchest?)

Reply:
+1 to making a toolchest rpm module.